matscipy/dislocation.py

```diff
@@ -2091,11 +2508,389 @@ def __init__(self, alat, C11, C12, C44, symbol='W'):
+        burgers = alat * np.array([1, -1, 0]) / 2.
+
+        disloCenterX = 0.5 * (alat * np.linalg.norm(axes[0])) / 6.0
+        # 1/4 + 1/2 (1/3 - 1/4) - to be in the middle of the glide set
+        disloCenterY = 7.0 * (alat * np.linalg.norm(axes[1])) / 24.0
+
+        unit_cell_core_position = np.array([disloCenterX,
+                                            disloCenterY, 0])
+
+        parity = [0, 0]
+
+        unit_cell = Diamond(symbol, directions=axes.tolist(),
+                            pbc=(False, False, True),
+                            latticeconstant=alat)
+
+        glide_distance = alat * np.linalg.norm(axes[0]) / 4.0
+
+        n_planes = 2
+
+        # 30 degree
+        burgers_left = alat * np.array([2., -1., -1.]) / 6.
+        left30 = DiamondGlide30degreePartial(alat, C11, C12, C44)
+        left30.set_burgers(burgers_left)
+        # another 30 degree
+        burgers_right = alat * np.array([1, -2, 1.]) / 6.
+        right30 = DiamondGlide30degreePartial(alat, C11, C12, C44)
+        self_consistent = False
+        super().__init__(left30, right30, unit_cell, alat, C11, C12, C44,
+                         axes, burgers, unit_cell_core_position, parity,
+                         glide_distance, n_planes=n_planes,
+                         self_consistent=self_consistent)
+
+
+class DiamondGlide60Degree(CubicCrystalDissociatedDislocation):
+    def __init__(self, alat, C11, C12, C44, symbol='C'):
+        axes = np.array([[1, 1, -2],
+                         [1, 1, 1],
+                         [1, -1, 0]])
+
+        # aiming for the resulting burgers vector
+        burgers = alat * np.array([1, 0, -1]) / 2.
+
+        disloCenterX = 0.5 * (alat * np.linalg.norm(axes[0])) / 6.0
+        # 1/4 + 1/2 (1/3 - 1/4) - to be in the middle of the glide set
+        disloCenterY = 7.0 * (alat * np.linalg.norm(axes[1])) / 24.0
+
+        unit_cell_core_position = np.array([disloCenterX,
+                                            disloCenterY, 0])
+
+        parity = [0, 0]
+
+        unit_cell = Diamond(symbol, directions=axes.tolist(),
+                            pbc=(False, False, True),
+                            latticeconstant=alat)
+
+        glide_distance = alat * np.linalg.norm(axes[0]) / 4.0
+
+        n_planes = 2
+
+        # 30 degree
+        burgers_left = alat * np.array([2., -1., -1.]) / 6.
+        left30 = DiamondGlide30degreePartial(alat, C11, C12, C44)
+        left30.set_burgers(burgers_left)
+        # 90 degree
+        burgers_right = alat * np.array([1, 1, -2.]) / 6.
+        right90 = DiamondGlide90degreePartial(alat, C11, C12, C44)
+        self_consistent = False
+        super().__init__(left30, right90, unit_cell, alat, C11, C12, C44,
+                         axes, burgers, unit_cell_core_position, parity,
+                         glide_distance, n_planes=n_planes,
+                         self_consistent=self_consistent)
+
+
+class FixedLineAtoms:
```

Comment:
is there not something like this in `ase.constraints` already?

Reply:
I thought so too, but it turned out that constraints available in `ase.constraints` are limited to some basic ones. I proposed to add my constraint to ase, but apparently they keep only basic constraints due to some deep design issues in `ase`. You can see a detailed explanation given by Ask [here](https://gitlab.com/ase/ase/-/issues/880). I think the take home message from that explanation is that one should add an instance of a constraint class to only one `Atoms` object or design an *immutable* constraint classes. In fact I should probably confirm this. I will ask Ask about it in the conversation of the ase issue referred above.

Reply:
OK, very happy to have it in `matscipy` for now.